src/pages/UserSettings/Teams.vue

```diff
@@ -0,0 +1,171 @@
+        await this.setCurrentTenant(removalTenant.slug)
+      }
+
+      const res = await this.$apollo.mutate({
+        mutation: require('@/graphql/Tenant/delete-membership.gql'),
+        variables: { membershipId }
+      })
+
+      if (res?.data?.delete_membership?.success) {
+        this.$emit(
+          'successful-action',
+          'You have removed yourself from this team.'
+        )
+      } else {
+        this.$emit(
+          'failed-action',
+          'Something went wrong while trying to leave this team. Please try again.'
+        )
+      }
+
+      if (initialTenant.id !== removalTenant.id) {
+        await this.setCurrentTenant(initialTenant.slug)
+      } else if (this.tenants[0].id === removalTenant.id) {
+        await this.setCurrentTenant(this.tenants[1].slug)
+      } else {
+        await this.setCurrentTenant(this.tenants[0].slug)
+      }
+
+      await this.getTenants()
+
+      this.isRemovingUser = false
+      this.dialogRemoveUser = false
+    },
+    async handleSwitchTenant(tenant) {
+      this.loading = true
+
+      if (tenant.slug == this.tenant.slug) return
+
+      await this.setCurrentTenant(tenant.slug)
+
+      this.loading = false
+
+      clearCache()
+      this.$router.push({
+        name: 'dashboard',
+        params: { tenant: this.tenant.slug }
+      })
+    }
+  }
+}
+</script>
+
+<template>
+  <ManagementLayout>
+    <template #title>Your Teams</template>
+
+    <template #subtitle>
+      See your Prefect teams and roles, switch between them, and leave teams
+    </template>
+
+    <v-data-table
+      fixed-header
+      class="elevation-2 rounded-none truncate-table"
+      :headers="headers"
+      :items="tenants"
+      :footer-props="{
+        showFirstLastPage: true,
+        firstIcon: 'first_page',
+        lastIcon: 'last_page',
+        prevIcon: 'keyboard_arrow_left',
+        nextIcon: 'keyboard_arrow_right'
+      }"
+    >
+      <template #item.role="{item}">
+        {{ role(item.role) }}
+      </template>
+      <template #item.id="{item}">
+        <v-btn v-if="item.id !== tenant.id" @click="handleSwitchTenant(item)"
+          >View</v-btn
+        >
+        <v-btn v-else disabled>Current</v-btn>
+        <v-btn
+          class="ml-3"
+          @click="
+            dialogRemoveUser = true
+            removeTenant = item
+          "
+          >Leave</v-btn
+        >
+        <ConfirmDialog
+          v-if="removeTenant"
+          v-model="dialogRemoveUser"
+          type="error"
+          :title="
+            `Are you sure you want to remove yourself from ${removeTenant.name}?`
+          "
+          :dialog-props="{ 'max-width': '600' }"
+          :disabled="isRemovingUser"
+          :loading="isRemovingUser"
+          @confirm="removeUser(removeTenant)"
+        >
+          <div class="red--text"
+            >Are you sure you want to remove yourself from
+            {{ removeTenant.name }}? You'll no longer be able to access your
```

Comment:
Maybe switch this to something like "run data associated with {{removeTenant.name}}"?  They should still be able to access their flow code etc.

Reply:
huh, this is the same wording used if you remove yourself from a team on the members page. If we want to update it, we should probably change it both places...

Reply:
updated in both locations!